aws/resource_aws_servicecatalog_product_test.go

```diff
@@ -0,0 +1,340 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAWSServiceCatalogProduct_basic(t *testing.T) {
+	salt := acctest.RandString(5)
+	resourceName := "aws_servicecatalog_product.test"
+	var describeProductOutput servicecatalog.DescribeProductAsAdminOutput
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsServiceCatalogProductDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSServiceCatalogProductConfig_basic(resourceName, "", "", salt, salt, salt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsServiceCatalogProductExists(resourceName, &describeProductOutput),
+					testAccCheckAwsServiceCatalogProductStandardFields(resourceName, &describeProductOutput, "product-"+salt),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "0"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSServiceCatalogProduct_disappears(t *testing.T) {
+	salt := acctest.RandString(5)
+	resourceName := "aws_servicecatalog_product.test"
+	var describeProductOutput servicecatalog.DescribeProductAsAdminOutput
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsServiceCatalogProductDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSServiceCatalogProductConfig_basic(resourceName, "", "", salt, salt, salt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsServiceCatalogProductExists(resourceName, &describeProductOutput),
+					testAccCheckAwsServiceCatalogProductDisappears(&describeProductOutput),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSServiceCatalogProduct_tags(t *testing.T) {
```

Comment:
see https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/contribution-checklists.md#resource-tagging-acceptance-testing-implementation

Reply:
removed the variables and now follows exactly the pattern in that post

Reply:
try to follow the example used in `TestAccAWSServiceCatalogPortfolio_Tags` for parity